bottlecap/src/lifecycle/invocation/processor.rs

```diff
@@ -103,11 +102,14 @@ impl Processor {
             config.trace_aws_service_representation_enabled,
         );
 
+        let enhanced_metrics = EnhancedMetrics::new(metrics_aggregator, Arc::clone(&config));
+        enhanced_metrics.start_enhanced_metrics_task(); // starts the long-running task that monitors usage metrics (fd_use, threads_use, tmp_used)
```

Comment:
I'd rename this to something like `start_usage_metrics_task` since it's already enhanced metrics(?)